src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs

```diff
@@ -50,29 +50,12 @@ internal class UnifiedSuggestedActionsSource
                 document, selection, includeSuppressionFixes, isBlocking,
                 addOperationScope, cancellationToken), cancellationToken).ConfigureAwait(false);
 
-            var filteredFixes = FilterOnAnyThread(fixes);
+            var filteredFixes = fixes.Where(c => c.Fixes.Length > 0).ToImmutableArray();
             var organizedFixes = OrganizeFixes(workspace, filteredFixes, includeSuppressionFixes);
 
             return organizedFixes;
         }
 
-        private static ImmutableArray<CodeFixCollection> FilterOnAnyThread(ImmutableArray<CodeFixCollection> collections)
-        {
-            return collections.Select(c => FilterIndividuallyOnAnyThread(c)).WhereNotNull().ToImmutableArray();
-        }
-
-        private static CodeFixCollection? FilterIndividuallyOnAnyThread(CodeFixCollection collection)
-        {
-            var applicableFixes = collection.Fixes;
-            return applicableFixes.Length == 0
-                ? null
-                : applicableFixes.Length == collection.Fixes.Length
```

Comment:
In case its hard to see, this condition is always true, and then the rest just falls away